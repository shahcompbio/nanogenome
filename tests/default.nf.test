nextflow_pipeline {

    name "Test pipeline"
    script "../main.nf"
    tag "pipeline"

    test("-profile test") {

        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            // stable_path: All files in ${params.outdir}/ with stable content
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            // bam_files: All bam files
            def bam_files  = getAllFilesFromDir(params.outdir, include: ['**/*.bam'])
            // vcf_files: All vcf files
            def vcf_files  = getAllFilesFromDir(params.outdir, include: ['**/*.vcf.gz', '**/*.vcf'])
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    // pipeline versions.yml file for multiqc from which Nextflow version is removed because we test pipelines on multiple Nextflow versions
                    removeNextflowVersion("$outputDir/pipeline_info/nanogenome_software_mqc_versions.yml"),
                    // All stable path name, with a relative path
                    stable_name,
                    // All files with stable contents
                    stable_path,
                    // All bam files
                    bam_files.collect { file -> file.getName() },
                    // All vcf files
                    vcf_files.collect { file -> file.getName() }
                ).match() }
            )
        }
    }
}
