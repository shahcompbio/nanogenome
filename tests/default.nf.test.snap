{
    "-profile test": {
        "content": [
            15,
            {
                "CLAIR3": {
                    "clair3": "1.2.0"
                },
                "LONGPHASE_PHASE": {
                    "longphase": 2.0
                },
                "NANOMONSV_GET": {
                    "nanomonsv": "0.8.0",
                    "mafft": 7.526,
                    "racon": "1.5.0",
                    "tabix": 1.21,
                    "bgzip": 1.21,
                    "python": "3.12.9"
                },
                "NANOMONSV_PARSE": {
                    "nanomonsv": "0.8.0",
                    "mafft": 7.526,
                    "racon": "1.5.0",
                    "tabix": 1.21,
                    "bgzip": 1.21,
                    "python": "3.12.9"
                },
                "SAMTOOLS_INDEX": {
                    "samtools": "1.22.1"
                },
                "SAVANA_CLASSIFY": {
                    "savana": "1.3.6"
                },
                "SEVERUS": {
                    "severus": 1.5
                },
                "TABIX_TABIX": {
                    "tabix": 1.21
                },
                "WAKHAN_CNA": {
                    "wakhan": "0.2.0"
                },
                "WAKHAN_HAPCORRECT": {
                    "wakhan": "0.2.0"
                },
                "WHATSHAP_HAPLOTAG": {
                    "whatshap": 2.8
                },
                "Workflow": {
                    "shahcompbio/nanogenome": "v1.0.0dev"
                }
            },
            [
                "clair3",
                "clair3/COLO829merge_output.vcf.gz",
                "clair3/COLO829merge_output.vcf.gz.tbi",
                "longphase",
                "longphase/COLO829.vcf.gz",
                "multiqc",
                "multiqc/multiqc_data",
                "multiqc/multiqc_data/multiqc.log",
                "multiqc/multiqc_data/multiqc_citations.txt",
                "multiqc/multiqc_data/multiqc_data.json",
                "multiqc/multiqc_data/multiqc_software_versions.txt",
                "multiqc/multiqc_data/multiqc_sources.txt",
                "multiqc/multiqc_report.html",
                "nanomonsv",
                "nanomonsv/COLO829",
                "nanomonsv/COLO829/COLO829.normal.bp_info.sorted.bed.gz",
                "nanomonsv/COLO829/COLO829.normal.bp_info.sorted.bed.gz.tbi",
                "nanomonsv/COLO829/COLO829.normal.deletion.sorted.bed.gz",
                "nanomonsv/COLO829/COLO829.normal.deletion.sorted.bed.gz.tbi",
                "nanomonsv/COLO829/COLO829.normal.insertion.sorted.bed.gz",
                "nanomonsv/COLO829/COLO829.normal.insertion.sorted.bed.gz.tbi",
                "nanomonsv/COLO829/COLO829.normal.rearrangement.sorted.bedpe.gz",
                "nanomonsv/COLO829/COLO829.normal.rearrangement.sorted.bedpe.gz.tbi",
                "nanomonsv/COLO829/COLO829.tumor.bp_info.sorted.bed.gz",
                "nanomonsv/COLO829/COLO829.tumor.bp_info.sorted.bed.gz.tbi",
                "nanomonsv/COLO829/COLO829.tumor.deletion.sorted.bed.gz",
                "nanomonsv/COLO829/COLO829.tumor.deletion.sorted.bed.gz.tbi",
                "nanomonsv/COLO829/COLO829.tumor.insertion.sorted.bed.gz",
                "nanomonsv/COLO829/COLO829.tumor.insertion.sorted.bed.gz.tbi",
                "nanomonsv/COLO829/COLO829.tumor.nanomonsv.result.txt",
                "nanomonsv/COLO829/COLO829.tumor.nanomonsv.result.vcf",
                "nanomonsv/COLO829/COLO829.tumor.nanomonsv.sbnd.result.txt",
                "nanomonsv/COLO829/COLO829.tumor.nanomonsv.supporting_read.txt",
                "nanomonsv/COLO829/COLO829.tumor.rearrangement.sorted.bedpe.gz",
                "nanomonsv/COLO829/COLO829.tumor.rearrangement.sorted.bedpe.gz.tbi",
                "nanomonsv/COLO829/versions.yml",
                "pipeline_info",
                "pipeline_info/nanogenome_software_mqc_versions.yml",
                "samtools",
                "samtools/COLO829.normal.haplotagged.bam.bai",
                "samtools/COLO829.tumor.haplotagged.bam.bai",
                "savana",
                "savana/COLO829",
                "savana/COLO829/COLO829.tumor.haplotagged.classified.somatic.vcf",
                "savana/COLO829/COLO829.tumor.haplotagged.inserted_sequences.fa",
                "savana/COLO829/COLO829.tumor.haplotagged.sv_breakpoints_read_support.tsv",
                "severus",
                "severus/COLO829",
                "severus/COLO829/all_SVs",
                "severus/COLO829/all_SVs/severus_all.vcf",
                "severus/COLO829/breakpoints_double.csv",
                "severus/COLO829/read_qual.txt",
                "severus/COLO829/severus.log",
                "severus/COLO829/somatic_SVs",
                "severus/COLO829/somatic_SVs/severus_somatic.vcf",
                "tabix",
                "tabix/rephased.vcf.gz.tbi",
                "wakhan",
                "wakhan/COLO829",
                "wakhan/COLO829/COLO829_heatmap_ploidy_purity.html",
                "wakhan/COLO829/COLO829_optimized_peak.html",
                "wakhan/COLO829/cna_solutions",
                "wakhan/COLO829/cna_solutions/solution_1",
                "wakhan/COLO829/cna_solutions/solution_1/COLO829_5.17_1.0_1.0_genes_genome.html",
                "wakhan/COLO829/cna_solutions/solution_1/COLO829_5.17_1.0_1.0_genome_copynumbers_breakpoints.html",
                "wakhan/COLO829/cna_solutions/solution_1/COLO829_5.17_1.0_1.0_genome_copynumbers_breakpoints_subclonal.html",
                "wakhan/COLO829/cna_solutions/solution_1/COLO829_5.17_1.0_1.0_genome_copynumbers_details.html",
                "wakhan/COLO829/cna_solutions/solution_1/bed_output",
                "wakhan/COLO829/cna_solutions/solution_1/bed_output/COLO829_5.17_1.0_1.0_copynumbers_segments_HP_1.bed",
                "wakhan/COLO829/cna_solutions/solution_1/bed_output/COLO829_5.17_1.0_1.0_copynumbers_segments_HP_2.bed",
                "wakhan/COLO829/cna_solutions/solution_1/bed_output/COLO829_5.17_1.0_1.0_copynumbers_subclonal_segments_HP_1.bed",
                "wakhan/COLO829/cna_solutions/solution_1/bed_output/COLO829_5.17_1.0_1.0_copynumbers_subclonal_segments_HP_2.bed",
                "wakhan/COLO829/cna_solutions/solution_1/bed_output/genes_copynumber_states.bed",
                "wakhan/COLO829/cna_solutions/solution_1/bed_output/loh_regions.bed",
                "wakhan/COLO829/cna_solutions/solution_1/bed_output/loh_regions_subclonal.bed",
                "wakhan/COLO829/cna_solutions/solution_1/variation_plots",
                "wakhan/COLO829/cna_solutions/solution_1/variation_plots/CN_VARIATION_INDEX.html",
                "wakhan/COLO829/cna_solutions/solution_1/variation_plots/chr22_cn.html",
                "wakhan/COLO829/cna_solutions/solution_1/vcf_output",
                "wakhan/COLO829/cna_solutions/solution_1/vcf_output/COLO829_5.17_1.0_1.0_wakhan_cna_integers.vcf",
                "wakhan/COLO829/cna_solutions/solution_1/vcf_output/COLO829_5.17_1.0_1.0_wakhan_cna_subclonals.vcf",
                "wakhan/COLO829/coverage_plots",
                "wakhan/COLO829/coverage_plots/COVERAGE_INDEX.html",
                "wakhan/COLO829/coverage_plots/chr22_cov.html",
                "wakhan/COLO829/hapcorrect_out",
                "wakhan/COLO829/hapcorrect_out/coverage_data",
                "wakhan/COLO829/hapcorrect_out/coverage_data/0_SNPs.csv",
                "wakhan/COLO829/hapcorrect_out/coverage_data/1_SNPs.csv",
                "wakhan/COLO829/hapcorrect_out/coverage_data/cancer_genes_coverage.csv",
                "wakhan/COLO829/hapcorrect_out/coverage_data/coverage.csv",
                "wakhan/COLO829/hapcorrect_out/coverage_data/coverage_ps.csv",
                "wakhan/COLO829/hapcorrect_out/coverage_data/phase_corrected_coverage.csv",
                "wakhan/COLO829/hapcorrect_out/coverage_data/pileup_SNPs.csv",
                "wakhan/COLO829/hapcorrect_out/data",
                "wakhan/COLO829/hapcorrect_out/data/0.bed",
                "wakhan/COLO829/hapcorrect_out/data/1.bed",
                "wakhan/COLO829/hapcorrect_out/data/COLO829.vcf_het_phased_snps.vcf.gz",
                "wakhan/COLO829/hapcorrect_out/data/COLO829.vcf_het_snps.csv",
                "wakhan/COLO829/hapcorrect_out/data/PAU59949.d052sup4305mCG_5hmCGvHg38.chr22_reduced.bam",
                "wakhan/COLO829/hapcorrect_out/data_phasing",
                "wakhan/COLO829/hapcorrect_out/data_phasing/COLO829.vcf_het_snps.csv",
                "wakhan/COLO829/hapcorrect_out/data_phasing/COLO829.vcf_het_snps_freqs.csv",
                "wakhan/COLO829/hapcorrect_out/data_phasing/COLO829.vcf_phasesets.csv",
                "wakhan/COLO829/hapcorrect_out/data_phasing/COLO829_phase_change_segments.csv",
                "wakhan/COLO829/hapcorrect_out/data_phasing/COLO829_phasesets.csv",
                "wakhan/COLO829/hapcorrect_out/data_phasing/snps_frequencies.csv",
                "wakhan/COLO829/hapcorrect_out/phasing_output",
                "wakhan/COLO829/hapcorrect_out/phasing_output/PHASE_CORRECTION_INDEX.html",
                "wakhan/COLO829/hapcorrect_out/phasing_output/chr22_phase_correction_0.1.html",
                "wakhan/COLO829/hapcorrect_out/phasing_output/chr22_phase_correction_0.2.html",
                "wakhan/COLO829/hapcorrect_out/phasing_output/chr22_phase_correction_0.3.html",
                "wakhan/COLO829/hapcorrect_out/phasing_output/chr22_phase_correction_1.html",
                "wakhan/COLO829/hapcorrect_out/phasing_output/chr22_phase_correction_2.html",
                "wakhan/COLO829/hapcorrect_out/phasing_output/chr22_without_phase_correction.html",
                "wakhan/COLO829/hapcorrect_out/phasing_output/rephased.vcf.gz",
                "wakhan/COLO829/hapcorrect_out/phasing_output/rephased.vcf.gz.csi",
                "wakhan/COLO829/hapcorrect_out/wakhan.log",
                "wakhan/COLO829/solutions_ranks.tsv",
                "wakhan/COLO829/versions.yml",
                "whatshap",
                "whatshap/COLO829.normal.haplotagged.bam",
                "whatshap/COLO829.tumor.haplotagged.bam"
            ],
            [
                "multiqc_citations.txt:md5,4c806e63a283ec1b7e78cdae3a923d4f",
                "COLO829.normal.bp_info.sorted.bed.gz:md5,b26f7a415fefec9917d591ca9ab46f3f",
                "COLO829.normal.bp_info.sorted.bed.gz.tbi:md5,1ffb1927d4e58d6c53ebf6e64f5d4daa",
                "COLO829.normal.deletion.sorted.bed.gz:md5,71dc1e3f5dc80679afd0c4172697e562",
                "COLO829.normal.deletion.sorted.bed.gz.tbi:md5,a97a9b7f3d3d6079a64aca6f0915947e",
                "COLO829.normal.insertion.sorted.bed.gz:md5,d424f16f1cb9670781f21f1b4214c6f4",
                "COLO829.normal.insertion.sorted.bed.gz.tbi:md5,8918dd5dbd98708c879facb094234004",
                "COLO829.normal.rearrangement.sorted.bedpe.gz:md5,45b5917895618e72da83cf81071bc537",
                "COLO829.normal.rearrangement.sorted.bedpe.gz.tbi:md5,4ff9a8c4b2337c26f9e225125278549a",
                "COLO829.tumor.bp_info.sorted.bed.gz:md5,bc1912a1d6f968b4d34fdfa99afe207d",
                "COLO829.tumor.bp_info.sorted.bed.gz.tbi:md5,1a48e978d0238ba0c66da5043a1f2d0c",
                "COLO829.tumor.deletion.sorted.bed.gz:md5,fa006fad34fd9040a6f6bdc91696f4b0",
                "COLO829.tumor.deletion.sorted.bed.gz.tbi:md5,556d95237f4f57df20a02f7636c77ae6",
                "COLO829.tumor.insertion.sorted.bed.gz:md5,3e25077c72c321293351e4a1f8e7a615",
                "COLO829.tumor.insertion.sorted.bed.gz.tbi:md5,7b7aca8272aa37837abf16108a425cc9",
                "COLO829.tumor.nanomonsv.result.txt:md5,3fdee28af0d5eaaaae5958647e6f0f09",
                "COLO829.tumor.nanomonsv.sbnd.result.txt:md5,1ed0a60fe6df23f9b8e0da253d45d4de",
                "COLO829.tumor.nanomonsv.supporting_read.txt:md5,acdb84ae7bb3e8032b38f817a5653681",
                "COLO829.tumor.rearrangement.sorted.bedpe.gz:md5,608cc47007f0835898122e822e684006",
                "COLO829.tumor.rearrangement.sorted.bedpe.gz.tbi:md5,131b771e01e94a5de15c6f48c1ef6467",
                "versions.yml:md5,80c71a07a01659756d2a33d0915d4d6c",
                "COLO829.tumor.haplotagged.inserted_sequences.fa:md5,f03f6e470d01d50b0b490a8003b5a66c",
                "COLO829.tumor.haplotagged.sv_breakpoints_read_support.tsv:md5,17740f351856e0d24093f78dcb73d68e",
                "breakpoints_double.csv:md5,ef54c710fe4b5f94bec1056c76729394",
                "read_qual.txt:md5,632b98d4b24bc2f357b6be55485b2f0d",
                "COLO829_5.17_1.0_1.0_copynumbers_segments_HP_1.bed:md5,7c11ed68d84a762d1ca9ef1cc2a06581",
                "COLO829_5.17_1.0_1.0_copynumbers_segments_HP_2.bed:md5,b33d2bfc1fbb366c55453cb98cd6eabd",
                "COLO829_5.17_1.0_1.0_copynumbers_subclonal_segments_HP_1.bed:md5,6192fc302d9e6e008f6e27af6faaa2a7",
                "COLO829_5.17_1.0_1.0_copynumbers_subclonal_segments_HP_2.bed:md5,a47b1142a74d920753ffc5e17a90deb1",
                "genes_copynumber_states.bed:md5,65f3390e4394954d39a8d115f583f749",
                "loh_regions.bed:md5,051d016d7f985f3fb0580ff0e1f1457b",
                "loh_regions_subclonal.bed:md5,051d016d7f985f3fb0580ff0e1f1457b",
                "0_SNPs.csv:md5,533eff72e31bf559a5a89ff7882b6f45",
                "1_SNPs.csv:md5,551fdfa6b9b1f493e4d922dbf72ebdf2",
                "cancer_genes_coverage.csv:md5,d41d8cd98f00b204e9800998ecf8427e",
                "coverage.csv:md5,3efb3125cc7ea789df91f3adc760b9bc",
                "coverage_ps.csv:md5,8c5371bcfdffb80103fe5d9d17565f70",
                "phase_corrected_coverage.csv:md5,9c30f72989f481c68b8a683e41c6609f",
                "pileup_SNPs.csv:md5,55a4bf41d63deb5fd60d2f4bd4220a3c",
                "0.bed:md5,4e33ed4ea1607432b4e7fbac899be3ed",
                "1.bed:md5,a4f7898d39aa2e7943f003249cb8128a",
                "COLO829.vcf_het_snps.csv:md5,6f6bf119c08382ca6a2c668aff642e7c",
                "COLO829.vcf_het_snps.csv:md5,015a8b262bfb8a7065f753668431d6d5",
                "COLO829.vcf_het_snps_freqs.csv:md5,593a00cd39838d150e245552a73fcbac",
                "COLO829.vcf_phasesets.csv:md5,9275a9dd47ea2dea6b222de0492fa275",
                "COLO829_phase_change_segments.csv:md5,5d29fe5f0cc55a7a30847659cfbc49fe",
                "COLO829_phasesets.csv:md5,a24e11e6936cc7b209e7f2de7a0b2474",
                "snps_frequencies.csv:md5,6e1907dae5172ef799b26e2a50f575bc",
                "solutions_ranks.tsv:md5,21794fac29813861cca2b344c8f0010b",
                "versions.yml:md5,43c0b9751153690bfedb3601de1d45ef"
            ],
            [
                "PAU59949.d052sup4305mCG_5hmCGvHg38.chr22_reduced.bam",
                "COLO829.normal.haplotagged.bam",
                "COLO829.tumor.haplotagged.bam"
            ],
            [
                "COLO829merge_output.vcf.gz",
                "COLO829.vcf.gz",
                "COLO829.tumor.nanomonsv.result.vcf",
                "COLO829.tumor.haplotagged.classified.somatic.vcf",
                "severus_all.vcf",
                "severus_somatic.vcf",
                "COLO829_5.17_1.0_1.0_wakhan_cna_integers.vcf",
                "COLO829_5.17_1.0_1.0_wakhan_cna_subclonals.vcf",
                "COLO829.vcf_het_phased_snps.vcf.gz",
                "rephased.vcf.gz"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.7"
        },
        "timestamp": "2025-10-07T22:47:27.569750169"
    }
}